# Advanced Data Science &amp; Python for Finance  <br><br> Final Project <br><br> Part 3 - Trading Strategy and Backtesting
-----

FIN580-59305 Spring 2020

Name: Danni Chen <br>
NetID: dannic3

Master of Science in Finance, University of Illinois at Urbana-Champaign.

* https://www.linkedin.com/in/dannie-chen-dc/

#### Instruction: Part 3 - Trading Strategy and Backtesting - Total Points: 20
For this section you will create a trading and backtesting strategy from scratch using pandas and analyze it. Finally, you will use backtrader to backtest a portfolio strategy based on the industry that you selected.

3.1. Use Intrinio API to pull the daily stock prices of all the constituents of your selected industry. Make sure that you have at least 100 trading days.

3.2 Select one company (stock) from your selected industry. In a short paragraph explain why you selected that particular company. For the company that you selected, create a long/short trading strategy. Make sure that you determine appropriate short and long trading windows and plot your strategy.

3.3 Backtest a $100,000 long/short portfolio using Pandas and plot your portfolio value.

3.4 Evaluate your trading strategy, calculate the portfolio sharpe ratio, maximum drawdown and Compound Annual Growth Rate (CAGR). In a short paragraph interpret the results.

3.5 Use backtrader to backtest a portfolio based on the industry that you selected. Make sure that there are executed trades and upload the plot that backtrader generated (use provided notebook)

Submit a clean and organized notebook and the corresponding dataset answering the questions for this part.

## 3.2 Select "Adobe Inc." from Application Software Industry for Furthur Analysis

In a short paragraph explain why you selected that particular company. For the company that you selected, create a long/short trading strategy. Make sure that you determine appropriate short and long trading windows and plot your strategy.

** Find companies meeting the following 3 conditions AT THE SAME TIME:
<br> 1. percentage price change in 30 days is positive
<br> 2. percentage price change YTD is positive
<br> 3. percentage price change in 12 month is positive**

In [18]:
import pandas as pd

In [19]:
sp1500_df = pd.read_csv("../../data/SP1500.csv")
software = sp1500_df[sp1500_df['industry']=='Application Software']

In [22]:
software_good = software[(software['pct_pricechange_30day']>0) & (software['pct_price_change_ytd']>0) & (software['pct_price_change_12_month']>0)]
software_good

company ticker  price_close  \
21                      Adobe Inc.   ADBE       343.84   
214   Cadence Design Systems, Inc.   CDNS        75.84   
289           Citrix Systems, Inc.   CTXS       142.58   
1232                Synopsys, Inc.   SNPS       149.64   
1358      Tyler Technologies, Inc.    TYL       314.78   

      pct_price_change_lastday  pct_pricechange_30day  pct_price_change_ytd  \
21                       -2.77                  14.15                  4.25   
214                      -6.52                  16.25                  9.34   
289                      -1.68                   2.47                 28.57   
1232                     -4.76                  18.80                  7.50   
1358                     -1.84                   7.96                  4.92   

      pct_price_change_12_month price_earnings_ratio price_bookvalue_ratio  \
21                        21.35               51.99x                15.87x   
214                       10.65               21.42x                 9.81x   
289                       41.48               24.90x                    NM   
1232                      24.61               48.05x                 5.42x   
1358                      39.34               75.79x                 7.31x   

      marketcap_mm revenue_mm  exchange              industry  
21       165662.40    11661.3  NasdaqGS  Application Software  
214       21298.39    2377.53  NasdaqGS  Application Software  
289       17601.59    3152.37  NasdaqGS  Application Software  
1232      22479.99    3374.67  NasdaqGS  Application Software  
1358      12516.17     1115.9      NYSE  Application Software

### Reason to Select Adobe Inc. : 
** Adobe is one of the five companies (listed above) which have good performance in both long term and short term, especially under this pandemic situation. This could be proved by the fact that the percentage price change of Adobe in 30 days, YTD, and in 12 months are 14.15%, 4.25%, and 21.35% repectively, which are all POSITIVE. <br>Although this year COVID-19 negatively affected many companies, Adobe still have positive returns year to date and within a month. So it might be regarded as a defensive business, which could still perform good even in bad economic situations. This might due to its price lies on a good fundamental basis and business nature. As we can see, the 12 month percentage change in price is 21.35, which is pretty high, implying that Adobe is doing good in long-run basis. <br>Also, the PE ratio of Adobe is 51.99x. In comparison, the 5 companies with good performance listed above have PE ratios from 21.42x to 75.79x. The current PE ratio of Adobe is good, which shows that investors are expecting high growth rates of Adobe in the future. **